### PR TITLE
introduce QA pipeline for protobuf schemas

### LINKED ISSUE #384: chore: add linter for protobuf schema files
we currntly do have linters for the XML schema.
we currntly do have linters for the JSON schema.
we currntly do **NOT** have linters for the ProtocolBuffer schema, don't we?

goal: introduce an automated tool chain, that can
- lint protocol buffer schema
  - find malformed schema files
  - find conflicts in field enumeration (duplicate use of a field number, etc ...)
  - optional but highly appreciated: detect breaking changes

this tooling should monitor
- all current (and future?) protobuff schema files in this repo's `schema` folder
  -  glob `schema/bom-*.proto` 

this tooling should run automatically on:
- each pullrequest
- each change in "master" and other relevant branches

the (github) workflow should be encapsulated standalone, not interfering any other/existing workflows

this tooling should be able to run locally

### PR DESCRIPTION
**Pull request by jkowalleck**
current protobuf schema files are not perfect.

this PR aims to prevent mistakes in the future, while acknowledging issues from the past. 

fixes #384

----

## status
- [x] introduce protobuf QA tools and configure them to our needs
- [x] baseline existing protobuf QA violations - as acknowledgement
- [x] introduce the protobuf QA tools in automated pipeline
- [x] introduce tools that detect and prevent breaking changes (BCD) in protobuf
- [x] introduce the protobuf BCD tools in automated pipeline
- [x] have our own protobuf test files checked against the schemas

## followup
- [ ] add the appropriate header to `*.textproto`  
  see https://github.com/CycloneDX/specification/issues/384#issuecomment-1959512503
- [ ] create ticket for BC: fix the proto3 schema enum value `0` -- they are intended to be fallbacks, not actual values.

### CODE REVIEW

**Comment by jkowalleck** [comment]
The schema linting and breaking-changes detection is in place.

@nscuro, could you do a review?

File: .github/workflows/test_proto.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# docs: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
+
+name: CT ProtoBuf
+
+on:
+  push:
+    branches: ['master', 'main']
```
Comment (nscuro) [inline .github/workflows/test_proto.yml:7]: There is no `main` branch here.

**Review by nscuro — APPROVED** [review]
Very well done!

File: .github/workflows/test_proto.yml (hunk 1 above)
Comment (jkowalleck) [reply .github/workflows/test_proto.yml:7]: yep. in case we ever go with the name change of the default branch from `master` to `main`, we are prepared.

**Comment by jkowalleck** [comment]
this is how a positive run report looks like : https://github.com/CycloneDX/specification/actions/runs/8081208041/job/22079240901
this is ow a negative run would look like: https://github.com/CycloneDX/specification/pull/388
 - see annotation in file around line 1791 -> https://github.com/CycloneDX/specification/pull/388/files#diff-31a634760e9b4432c392ead00601567422a8cc12ac462dead1a7f7ab9fa90fdbR1791 
 - see run res: https://github.com/CycloneDX/specification/actions/runs/8019818776/job/21908366440#step:3:31

**Pull request merged**